src/components/app/app.js

```diff
@@ -52,13 +52,27 @@ const inputModeFromHotKey = {
     v: 'color',
 }
 
-function initialGridFromURL () {
+function determineInitialGrid () {
     const params = new URLSearchParams(window.location.search);
-    const grid = newSudokuModel({
+    let grid = newSudokuModel({
         initialDigits: params.get('s'),
         difficultyLevel: params.get('d'),
-        storeCurrentSnapshot: sn => document.body.dataset.currentSnapshot = sn,
+        onGamestateChange: grid => {
+            // we must persist this for use by bookmarklets
+            document.body.dataset.currentSnapshot = grid.get('currentSnapshot');
+            modelHelpers.persistGamestate(grid);
+        }
     });
+
+    if (params.get('r') === "1") {
```

Comment:
`r` is the flag that tells the app we should attempt to load a grid from local storage.